subprojects/core/src/test/groovy/org/gradle/api/internal/file/archive/TarFileTreeTest.java

```diff
@@ -159,14 +159,15 @@ public void readsTarFileWithNullPermissions() {
     }
 
     @Test
-    public void doesNotOverwriteFilesOnSecondVisit() {
+    public void doesNotOverwriteFilesOnSecondVisit() throws InterruptedException {
         rootDir.file("file1.txt").write("content");
         rootDir.tarTo(tarFile);
 
         assertVisits(tree, toList("file1.txt"), new ArrayList<String>());
-        File unpackedFile = expandDir.listFiles()[0].listFiles()[0];
-        unpackedFile.setWritable(false);
+        TestFile content = expandDir.listFiles()[0].listFiles()[0];
+        TestFile.Snapshot snapshot = content.snapshot();
+        Thread.sleep(1000);
```

Comment:
You should instead call `unpackedFile.makeOlder()`. That would imply converting this test to Spock first.